File: Hrms.Web/src/app/admin/directory/models/employee.model.ts
import { AddressModel } from "src/app/shared/models/address.model";
import { DepartmentModel, DesignationModel } from "src/app/admin/company/models";
import { FileDetailModel } from "src/app/employee/profile/models";

export class EmployeeModel {
  id: number;
  name: string;
  firstName: string;
  lastName: string;
  employeeName: string;
  dateOfBirth: string;
  gender: number;
  bloodGroup: number;
  maritalStatus: number;
  email: string;
  alternateEmail: string;
  phone: number;
  alternatePhone: number;
  employeeID: string;
  dateOfJoining: string;
  dateOfLeaving: string;
  companyId: number;
  probationPeriod: number;
  workExperience: string;
  designationId: number;
  designationName: string;
  jobTitle: string;
  departmentId: number;
  departmentName:string;
  subDepartment: string;
  fromDate: string;
  toDate: string;
  employeeType: number;
  relationship: string;
  qualificationType: string;
  universityName: string;
  collegeName: string;
  courseType: string;
  courseName: string;
  stream: string;
  courseStartDate: string;
  courseEndDate: string;
  uploadedBy: string;
  status: number;
  currentAddress: AddressModel;
  permanentAddress: AddressModel;
  department: DepartmentModel;
  designation: DesignationModel;
  code: string
  exitDate: string;
  noticePeriod: number;
  note: string;
  exitType: number;
  startedFrom: string;
  roleId: number;
  imageUrl:string;
  imageDetails:FileDetailModel;
  
  constructor() {
    this.currentAddress = new AddressModel();
    this.permanentAddress = new AddressModel();
    this.department = new DepartmentModel();
    this.designation = new DesignationModel();
    this.noticePeriod = 0;
    this.imageDetails = new FileDetailModel();
  }
}

File: Hrms.Core/Models/Employee/EmployeeModel.cs
using Hrms.Core.Models.Company;
using Hrms.Core.Models.JobApplication;
using Hrms.Core.Utilities;


namespace Hrms.Core.Models.Employee
{
    public class EmployeeModel
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public int UserId { get; set; }
        public int CompanyId { get; set; }
        public int? DesignationId { get; set; }
        public string DesignationName { get; set; }
        public int? DepartmentId { get; set; }
        public string DepartmentName { get; set; }
        public int? ManagerId { get; set; }
        public string ManagerName { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmployeeName { get; set; }
        public int BloodGroup { get; set; }
        public int MaritalStatus { get; set; }
        public int Gender { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string AlternatePhone { get; set; }
        public string AlternateEmail { get; set; }

        public DateTime? DateOfBirth { get; set; }
        public DateTime DateOfJoining { get; set; }
        public DateTime? DateOfLeaving { get; set; }
        public DateTime? StartedFrom { get; set; }
        public Constants.RecordStatus Status { get; set; }

        public AddressModel PermanentAddress { get; set; }
        public AddressModel CurrentAddress { get; set; }
        public string  Department { get; set; }
        public string Designation { get; set; }

        public int ProbationPeriod { get; set; }
        public int EmployeeType { get; set; }
        public int RoleId { get; set; }
        public DateTime? ExitDate { get; set; }
        public int? ExitType { get; set; }
        public int NoticePeriod { get; set; }
        public string Note { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreatedOn { get; set; }
        public FileDetailModel ImageDetails { get; set; }

    }
}
